core/job/service/job_service.go

```diff
@@ -1915,3 +1962,27 @@ func (j *JobService) GetDownstreamByResourceURN(ctx context.Context, tnnt tenant
 
 	return dependentJobs, nil
 }
+
+func (j *JobService) getDownstreamJobs(ctx context.Context, projectName tenant.ProjectName, jobs []*job.Job) ([]*job.Job, error) {
```

Comment:
seems this function only returns inferred downstream (which I believe is sufficient). can you rename the function accordingly?

Reply:
just aware about this, it should be not only inferred downstream but also the static one, I think we miss this case

Reply:
let's rethink whether it is necessary. we probably don't need to update the static downstream, because the reason why we need to refresh the existing/new downstream is because the table `destination` has changed.

Reply:
got it, noted